Remove link to Delete Card interface on some parts of buy and management flows

brampitoyo opened the pull request:
When you’re buying an item, you want to be focused on selecting a card
or adding a new one, not on deleting.

On certain parts of the management interface (e.g., select card for a
subscription), you don’t want to delete a card, either. You just want
to select. Maybe you’d like to add a card, but definitely not delete.

andymckay commented:
I couldn't spot what changed in the first two files, but the simplified purchase flow looks right to me. Thanks. r+

brampitoyo commented:
On pay flow, and on the change payment interface on management flow, I’ve removed “Add” and “Delete” links. Then I updated the flow. That’s about it!
